lib/bundler/definition.rb

```diff
@@ -706,9 +711,6 @@ def converge_dependencies
         elsif dep.source
           dep.source = sources.get(dep.source)
         end
-        if dep.source.is_a?(Source::Gemspec)
-          dep.platforms.concat(@platforms.map {|p| Dependency::REVERSE_PLATFORM_MAP[p] }.flatten(1)).uniq!
-        end
```

Comment:
This is a part of reverting 0a8ca4879e0b79aa4109a0dc424940b079ef38d0.
See also: #4150 and #4102.